src/execution/operator/persistent/physical_export.cpp

```diff
@@ -147,6 +163,7 @@ void PhysicalExport::GetData(ExecutionContext &context, DataChunk &chunk, Global
 	WriteCatalogEntries(ss, tables);
 	WriteCatalogEntries(ss, views);
 	WriteCatalogEntries(ss, indexes);
+	WriteCatalogEntries(ss, macros);
```

Comment:
Macro's can reference tables/views so there is no right order. I think we should figure out some kind of dependency, or order them all by `oid`. I tried ordering by timestamp, but that didn't work out.

Reply:
I was assuming that macros were entirely untyped and would fail dynamically but that's not right. E.g.

```sql
CREATE MACRO addn(n) AS (SELECT x + n FROM nonexistanttable);
```

gives  `Catalog Error: Table with name nonexistanttable does not exist!`.